### PR TITLE
add __setstate__ for BatchNorm to load old checkpoints

### LINKED ISSUE #5881: BatchNorm checkpoints are broken with the latest master
https://github.com/pytorch/pytorch/pull/5766 broke loading of old checkpoints.
Fix incoming

### PR DESCRIPTION
**Pull request by soumith**
fixes https://github.com/pytorch/pytorch/issues/5881

### CODE REVIEW

File: torch/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -64,6 +64,16 @@ def forward(self, input):
...
 
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+        if not hasattr(self, 'track_running_stats'):
+            self.track_running_stats = True
+        if not hasattr(self, 'num_batches_tracked'):
+            if self.track_running_stats:
+                self.register_buffer('num_batches_tracked', torch.LongTensor([0]))
+            else:
+                self.register_parameter('num_batches_tracked', None)
```
Comment (apaszke) [inline torch/nn/modules/batchnorm.py]: This needs to be `register_buffer` too!!

**Review by apaszke — APPROVED** [review]
